src/bam_writer.cpp

```
@@ -75,24 +63,20 @@ bam_writer_t *bam_writer_open(const char *path, const bntseq_t *bns,
     if (!idx_has_hd && !user_has_hd && compat->emit_hd &&
         sam_hdr_add_lines(hdr, compat->hd_line, 0) < 0) goto fail;
     if (!idx_has_sq) {
+        // One shared formatter (bwa.cpp) rather than sam_hdr_add_line varargs:
+        // the @SQ contents -- including AH:* on ALT contigs -- are identical on
+        // every path, and this writer having its own spelling is how it came to
+        // silently drop AH for every ALT-aware reference (#281). Not gated on
+        // `compat`: both upstreams emit this block, so it is correct output,
+        // not a divergence.
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)